contents/javascript/Javascript-framework-React.md

```diff
@@ -0,0 +1,142 @@
+<frontmatter>
+  title: React
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# React
+
+Author: [Aadyaa Maddi](https://github.com/amad-person)
+
+Reviewers: [Amrut Prabhu](https://github.com/amrut-prabhu), [Marvin Chin](https://github.com/marvinchin)
+
+### What is React?
+
+**React** is an open-source JavaScript <tooltip content="React only handles the view component of your application. It is not an MVC framework!">library</tooltip> for building composable user interfaces. It was developed by Facebook and released to the world in 2013, and is now used in a huge number of popular applications, including Facebook and Instagram.
+
+<box type="tip">
+    If you have the <a href="https://github.com/facebook/react-devtools">React DevTools</a> extension installed in your browser, you can inspect the webpage of either of these applications and observe that React is being used by them! 
+</box>
```

Comment:
This tip is probably not needed? I don't think anybody will doubt your claim that Facebook/Instagram is using React. After all, the three are from the same parent company.

Reply:
I actually wanted to show that React is a popular front-end library, and that React DevTools is a good way for readers to see that for themselves. 

> If you have the <a href="https://github.com/facebook/react-devtools">React DevTools</a> extension installed in your browser, you can inspect the webpages of [many popular applications](https://github.com/facebook/react/wiki/Sites-Using-React) (e.g. Airbnb, Netflix, etc.) and observe that React is being used by them!

Would this tip be okay?

Reply:
I actually wanted to introduce React DevTools in the tip box. I have rephrased the introductory sentence and modified the contents of the tip box to reflect this:

> **React** is an open-source JavaScript library for <tooltip content="React only handles the UI component of your application. It is not a full web application framework!">building the UI of web apps</tooltip>. It was developed by Facebook and released to the world in 2013, and is now used in a [large number of popular applications](https://github.com/facebook/react/wiki/Sites-Using-React), including Facebook and Instagram.

> The <a href="https://github.com/facebook/react-devtools">React DevTools</a> browser extension can be used for inspecting the webpages of these applications to see how React is being used by them.

Would this be better?

Reply:
Isn't it too early in the article to expect the reader to install a dev tool in the browser? The user has ready only two sentences so far.

Reply:
Good point, I'll remove the link or shift it to `Where to Go From Here` if needed.